### PR TITLE
[Basic Action Refactor] (No.5) Remove action functional interfaces.

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/action/CreateFunctionAppAction.java [hunk 1]
```diff
@@ -43,19 +43,20 @@
...
-import com.microsoft.tooling.msservices.serviceexplorer.listener.ActionBasicable;
 import rx.Single;
 
 import java.util.Objects;
 import java.util.function.Consumer;
 
 import static com.microsoft.intellij.ui.messages.AzureBundle.message;
 
-public class CreateFunctionAppAction extends NodeActionListener implements ActionBasicable {
+@Name("Create")
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/action/CreateFunctionAppAction.java:58]: Shall we use a new annotation like `Action`? As @Name may be a bit confusing

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/action/CreateFunctionAppAction.java:58]: good idea. we can consider to do that later. but not in this scope.

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java [hunk 2]
```diff
@@ -94,12 +93,9 @@ public BasicActionBuilder withPromptable(final boolean promptRequired) {
...
     }
 
-    /**
-     * TODO (Qianjin) : remove after telemetry is implemented in @AzureOperation.
-     */
-    public BasicActionBuilder withTelemetrable(final boolean telemetryRequired, TelemetryParameter telemetryParameter) {
+    public BasicActionBuilder withTelemetrable(final boolean telemetryRequired, String actionString) {
         this.telemetrable = telemetryRequired;
-        this.telemetryParameter = telemetryParameter;
+        this.actionString = actionString;
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java:98]: Could we just pass the `ActionConstants`?

**Review by Flanker32 — APPROVED** [review]

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java [hunk 3]
```diff
@@ -50,7 +49,7 @@
     private boolean promptable;
 
     private boolean telemetrable;
-    private TelemetryParameter telemetryParameter;
+    private String actionString;
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java:52]: May need to find a better name for this parameter, it contains the service & operation name, but we can not get the hint from the name

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**